tool.ts

```diff
@@ -153,28 +159,37 @@ export function findLocalToolVersions(toolName: string, arch?: string) {
 export async function downloadTool(url: string, fileName?:string): Promise<string> {
     return new Promise<string>(async (resolve, reject) => {
         try {
-            debug(fileName);
+            tl.debug(fileName);
             fileName = fileName || uuidV4();
             var destPath = path.join(_getAgentTemp(), fileName);
 
-            debug('downloading', url);
-            debug('destination', destPath);
+            tl.debug('downloading ' + url);
+            tl.debug('destination ' + destPath);
 
             if (fs.existsSync(destPath)) {
                 throw new Error("Destination file path already exists");
             }
 
             // TODO: retries
-            debug('creating stream');
+            tl.debug('creating stream');
             let file: NodeJS.WritableStream = fs.createWriteStream(destPath);
             file.on('open', async(fd) => {
-                debug('downloading');
-                let stream = (await http.get(url)).message.pipe(file);
-
-                stream.on('finish', () => {
-                    debug('download complete');
-                    resolve(destPath);
-                });
+                try {
+                    tl.debug('downloading');
+                    let response: httpm.HttpClientResponse = await http.get(url);
+                    if (response.message.statusCode != 200) {
+                        throw(new Error('Unexpected HTTP response: ' + response.message.statusCode));
+                    }
+
+                    let stream = response.message.pipe(file);
+                    stream.on('finish', () => {
+                        tl.debug('download complete');
+                        resolve(destPath);
+                    });
+                }
+                catch (err) {
+                    reject(err);
+                }
```

Comment:
non 200 was failing, but the task wasnt getting failed. i added a throw and also a try catch-and-reject in case anything else blows up inside this function.